crypto/rand/rand_lib.c

```diff
@@ -49,13 +72,19 @@ const RAND_METHOD *RAND_get_rand_method(void)
 #endif
             default_RAND_meth = RAND_OpenSSL();
     }
-    return default_RAND_meth;
+    tmp_meth = default_RAND_meth;
```

Comment:
Does default_RAND_meth have to be declared volatile for this to work? Can the compiler "optimise" this solution away? I'm not really sure.